modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -476,18 +498,56 @@ class SliceLayerImpl : public SliceLayer
         CV_TRACE_FUNCTION();
         CV_TRACE_ARG_VALUE(name, "name", name.c_str());
 
-        CV_OCL_RUN(IS_DNN_OPENCL_TARGET(preferableTarget),
-                   forward_ocl(inputs_arr, outputs_arr, internals_arr))
+        bool hasSteps = false;
```

Comment:
Perhaps this check should be moved out of `.forward()` method and calculated earlier (store value as class field).
Try to put that before `supportBackend()` callback to properly support for other backends: IE, Vulkan, CUDA or OpenCL target.